frontend/app/mocks/client-application-api.server.ts

```diff
@@ -0,0 +1,245 @@
+import { HttpResponse, http } from 'msw';
+import { z } from 'zod';
+
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('client-application-api.server');
+
+/**
+ * Server-side MSW mocks for the Client Application API
+ */
+export function getClientApplicationApiMockHandlers() {
+  log.info('Initializing Client Application mock handlers');
+
+  return [
+    http.post('https://api.example.com/v1/client-application', async ({ request }) => {
+      log.debug('Handling request for [%s]', request.url);
+      const statusRequestSchema = z.object({
```

Comment:
Minor nitpick: Consider renaming `statusRequestSchema` to something else since this mock is unrelated to statuses (same with `statusRequest` below)

(Same comment for the other mock endpoint)